File: CourseProjects/CourseProjects/VisualStudioProjects/Agile/MockingWithMoq/MockingChallengeStart/TravelAllowanceCalculatorTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace MockingChallengeFinish {

    [TestClass]
    public class TravelAllowanceCalculatorTest {
        [TestInitialize]
        public void Start() {
            calculator = new TravelAllowanceCalculator(null, null);
        }
        [TestMethod]
	    [Ignore]
        public void LocalFlightFirstClass() {
            Flight f = new Flight("Belfast", "Dublin", Seating.FIRST, DateTime.Now);
            Assert.AreEqual(123, calculator.CalculateAllowance(f), 0.001);
        }
        [TestMethod]
	    [Ignore]
        public void LocalFlightBusinessClass() {
            Flight f = new Flight("Belfast", "Dublin", Seating.BUSINESS, DateTime.Now);
            Assert.AreEqual(123, calculator.CalculateAllowance(f), 0.001);
        }
        [TestMethod]
	    [Ignore]
        public void LocalFlightEconomyClass() {
            Flight f = new Flight("Belfast", "Dublin", Seating.ECONOMY, DateTime.Now);
            Assert.AreEqual(123, calculator.CalculateAllowance(f), 0.001);
        }
        [TestMethod]
	    [Ignore]
        public void InternationalFlightFirstClass() {
            Flight f = new Flight("Heathrow", "Dublin", Seating.FIRST, DateTime.Now);
            Assert.AreEqual(123, calculator.CalculateAllowance(f), 0.001);
        }
        [TestMethod]
	    [Ignore]
        public void InternationalFlightBusinessClass() {
            Flight f = new Flight("Heathrow", "Dublin", Seating.BUSINESS, DateTime.Now);
            Assert.AreEqual(123, calculator.CalculateAllowance(f), 0.001);
        }
        [TestMethod]
	    [Ignore]
        public void InternationalFlightEconomyClass() {
            Flight f = new Flight("Heathrow", "Dublin", Seating.ECONOMY, DateTime.Now);
            Assert.AreEqual(123, calculator.CalculateAllowance(f), 0.001);
        }
        [TestMethod]
	    [Ignore]
	    public void MultipleLocalFlights() {
            Flight f1 = new Flight("Belfast", "Dublin", Seating.FIRST, DateTime.Now);
            Flight f2 = new Flight("Belfast", "Dublin", Seating.FIRST, DateTime.Now);
            Assert.AreEqual(1285, calculator.CalculateAllowance(f1, f2), 0.001);
	    }
        [TestMethod]
        [Ignore]
	    //Multiple international flights should qualify for extra allowance
	    public void MultipleInternationalFlights() {
            Flight f1 = new Flight("Heathrow", "Dublin", Seating.FIRST, DateTime.Now);
            Flight f2 = new Flight("Heathrow", "Dublin", Seating.FIRST, DateTime.Now);
		    Flight f3 = new Flight("Heathrow","Dublin",Seating.FIRST,DateTime.Now);
		    Assert.AreEqual(123,calculator.CalculateAllowance(f1,f2,f3),0.001);
	    }
        private TravelAllowanceCalculator calculator;
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/UnitTestingExercises/src/exercise/tdd/mocking/start/TravelAllowanceCalculatorTest.java
package exercise.tdd.mocking.start;

import static org.junit.Assert.*;
import static exercise.tdd.mocking.start.Seating.*;
import java.util.Date;

import org.junit.*;

import exercise.tdd.mocking.finish.jmock.FlightConcessionsEngine;
import exercise.tdd.mocking.finish.jmock.FlightPathDatabase;

public class TravelAllowanceCalculatorTest {
	@Before
	public void start() {
		calculator = new TravelAllowanceCalculator(null,null);
	}
	@Test
	@Ignore
    public void localFlightFirstClass() {
        Flight f = new Flight("Belfast","Dublin",FIRST,new Date());
        assertEquals(123,calculator.calculateAllowance(f),0.001);
    }
	@Test
	@Ignore
    public void localFlightBusinessClass() {
        Flight f = new Flight("Belfast","Dublin",BUSINESS,new Date());
        assertEquals(123,calculator.calculateAllowance(f),0.001);
    }
	@Test
	@Ignore
    public void localFlightEconomyClass() {
        Flight f = new Flight("Belfast","Dublin",ECONOMY,new Date());
        assertEquals(123,calculator.calculateAllowance(f),0.001);
    }
	@Test
	@Ignore
    public void internationalFlightFirstClass() {
        Flight f = new Flight("Heathrow","Dublin",FIRST,new Date());
        assertEquals(123,calculator.calculateAllowance(f),0.001);
    }
	@Test
	@Ignore
    public void internationalFlightBusinessClass() {
        Flight f = new Flight("Heathrow","Dublin",BUSINESS,new Date());
        assertEquals(123,calculator.calculateAllowance(f),0.001);
    }
	@Test
	@Ignore
    public void internationalFlightEconomyClass() {
        Flight f = new Flight("Heathrow","Dublin",ECONOMY,new Date());
        assertEquals(123,calculator.calculateAllowance(f),0.001);
    }
	@Test
	@Ignore
	public void multipleLocalFlights() {
		Flight f1 = new Flight("Belfast","Dublin",FIRST,new Date());
		Flight f2 = new Flight("Belfast","Dublin",FIRST,new Date());
		 assertEquals(1285,calculator.calculateAllowance(f1,f2),0.001);
	}
	@Test
	@Ignore
	//Multiple international flights should qualify for extra allowance
	public void multipleInternationalFlights() {
		Flight f1 = new Flight("Heathrow","Dublin",FIRST,new Date());
		Flight f2 = new Flight("Heathrow","Dublin",FIRST,new Date());
		Flight f3 = new Flight("Heathrow","Dublin",FIRST,new Date());
		assertEquals(123,calculator.calculateAllowance(f1,f2,f3),0.001);
	}
	private TravelAllowanceCalculator calculator;
	private FlightConcessionsEngine concessionsEngine;
	private FlightPathDatabase pathDatabase;
}
